src/aspire/orientation/__init__.py

```diff
@@ -363,6 +367,7 @@ def _get_shift_equations_approx(self, equations_factor=1, max_memory=4000):
             shift_eq[idx] = -1 * coeffs if is_pf_j_flipped else coeffs
 
         # create sparse matrix object only containing non-zero elements
+        # Note, this line is particularly sensitive to precision
```

Comment:
That's very strange, considering that all the arrays going into it can be `float32`. Is there some convergence issue later on that's causing problems? (According to the [docs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.lsqr.html), `lsqr` seems to have a hardcoded tolerance of `1e-8`, which might be the culprit.)

Reply:
I was a little worried what we would go on to do with that matrix might be sensitive... but it is true that the results follow the same level of tolerance seen elsewhere in that test (1e-6 for singles), hopefully that will satisfy the other platforms via CI too. pushing.

I added some lines that will log out the lsqr numerical summary for DEBUG level (so to the log file and for `pytest -s` atm).

Reply:
Ok good.